messaging/src/test/java/org/axonframework/eventhandling/ReplayTokenTest.java

```diff
@@ -135,4 +154,1470 @@ void createReplayTokenReturnsReplayToken() {
+
+            TrackingToken result = replayToken.advancedTo(newToken);
+
+            // Should still be in replay mode
+            assertInstanceOf(ReplayToken.class, result, "Should still be in replay mode");
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event 4 was processed before reset (it was NOT in the gap), " +
+                            "so it SHOULD be marked as replay");
+        }
+
+        @Test
+        void gapsNotFilledDuringReplay() {
+            // Setup: tokenAtReset at index 5 with gap at 3
+            GapAwareTrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(5, setOf(3L));
+
+            // During replay, gap 3 is NOT filled - we skip from 2 to 4
+            // After processing events 0,1,2 and then 4, currentToken has gap at 3
+            GapAwareTrackingToken currentTokenWithGap = GapAwareTrackingToken.newInstance(4, setOf(3L));
+
+            // Create replay token with this state
+            ReplayToken replayToken = new ReplayToken(tokenAtReset, currentTokenWithGap, null);
+
+            // Now event 5 arrives - this event WAS processed before reset
+            GapAwareTrackingToken newToken = GapAwareTrackingToken.newInstance(5, setOf(3L));
+
+            TrackingToken result = replayToken.advancedTo(newToken);
+
+            assertInstanceOf(ReplayToken.class, result, "Should still be in replay mode");
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event 5 was processed before reset, correctly marked as replay when gaps match");
+        }
+
+        @Test
+        void gapFilledThenEventAtResetIndex() {
+            // Setup: tokenAtReset at index 5 with gap at 3
+            // Before reset, processor saw: 0,1,2,4,5 (NOT 3)
+            GapAwareTrackingToken tokenAtReset = GapAwareTrackingToken.newInstance(5, setOf(3L));
+
+            // After processing 0,1,2,3,4 - gap filled, currentToken at 4 with no gaps
+            GapAwareTrackingToken currentToken = GapAwareTrackingToken.newInstance(4, emptySet());
+
+            ReplayToken replayToken = new ReplayToken(tokenAtReset, currentToken, null);
+
+            // Now event 5 arrives - this event WAS processed before reset
+            GapAwareTrackingToken newToken = GapAwareTrackingToken.newInstance(5, emptySet());
+
+            TrackingToken result = replayToken.advancedTo(newToken);
+
+            // Event 5 at the reset index should STILL be a replay because it was processed before.
+            assertTrue(ReplayToken.isReplay(result),
+                    "Event 5 was processed before reset, should be marked as replay");
+            assertInstanceOf(ReplayToken.class, result,
+                    "Should still be in replay mode at reset index - " +
+                            "event 5 was processed before reset");
+        }
+
+        @Test
+        void multipleGapsFilledCausesIncorrectReplayStatus() {
```

Comment:
I don't think this method name aligns with what's being test, nor do the comments within this test method block.